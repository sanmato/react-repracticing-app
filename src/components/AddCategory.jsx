import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onFormSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    // setCategories((categories) => [...categories, inputValue]);
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={(event) => onFormSubmit(event)}>
      <input
        type="text"
        placeholder="Buscar GIFs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};
