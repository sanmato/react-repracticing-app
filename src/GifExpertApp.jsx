import React, { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Kimetsu no Yaiba"]);

  const onAddCategory = (category) => {
    if (categories.includes(category)) return;
    setCategories([...categories, category]);
  };

  return (
    <div>
      <h1>GifExpertApp</h1>
      <AddCategory onNewCategory={(value) => onAddCategory(value)} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </div>
  );
};
