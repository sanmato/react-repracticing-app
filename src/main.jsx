import { createRoot } from "react-dom/client";
import { GifExpertApp } from "./GifExpertApp";
import "./styles.css";

createRoot(document.getElementById("root")).render(<GifExpertApp />);
