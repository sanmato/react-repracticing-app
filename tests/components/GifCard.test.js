import { GifCard } from "../../src/components";
import { render, screen } from "@testing-library/react";

describe("Pruebas en GifCard Snapshot", () => {
  const title = "Prueba 1";
  const url = "https://localhost.com/prueba1";
  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<GifCard title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de tener un parrafo con el title", () => {
    render(<GifCard title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("debe de tener la imagen igual al url y alt de los props", () => {
    render(<GifCard title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });
});
