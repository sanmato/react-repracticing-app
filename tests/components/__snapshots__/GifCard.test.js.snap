// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifCard Snapshot debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Prueba 1"
      src="https://localhost.com/prueba1"
    />
    <p>
      Prueba 1
    </p>
  </div>
</div>
`;
